core/src/main/scala/higherkindness/mu/rpc/srcgen/avro/AvroSrcGenerator.scala

```diff
@@ -152,40 +138,65 @@ final case class AvroSrcGenerator(
            )
          } else Nil)
 
-    val serviceParams: String = (serializationType.toString +: extraParams).mkString(",")
+    val serviceParams = (serializationType.toString +: extraParams).mkString(",")
+
+    val requestLines = {
+      val result = protocol.getMessages.asScala.toList.traverse {
+        case (name, message) =>
+          val comment = Seq(Option(message.getDoc).map(doc => s"  /** $doc */")).flatten
+          buildMethodSignature(name, message.getRequest, message.getResponse).map { content =>
+            comment ++ Seq(content, "")
+          }
+      }
+      result.map(_.flatten)
+    }
```

Comment:
If you use `List` everywhere instead of `Seq`, you can use `flatTraverse` instead of `traverse`, meaning you don't need the `.map(_.flatten)` at the end:

```scala
    val requestLines = protocol.getMessages.asScala.toList.flatTraverse {
      case (name, message) =>
        val comment = Option(message.getDoc).map(doc => s"  /** $doc */").toList
        buildMethodSignature(name, message.getRequest, message.getResponse).map { content =>
          comment ++ List(content, "")
        }
    }
```

Reply:
Yeah I remember trying to get this to work with `flatTraverse` but it makes sense that it couldn't handle the mixture of `List` and `Seq`s.  Thanks for the tip!